Add HISTORY.md entry for change noted in #39589

sostock opened the pull request:
As noted in https://github.com/JuliaLang/julia/issues/39589#issuecomment-776389595, it could be good to mention this change in NEWS/HISTORY. This adds a HISTORY.md item. I will also make a PR backporting this to the 1.6 NEWS.md file.

Edit: backport PR: #39785

sostock committed:
Add HISTORY.md entry for change noted in #39589

timholy commented:
I think it would be best to frame this as a bugfix from the beginning. Phrases like "now always describe" and "is considered a bugfix" make it sound like a bit of a dicey change to be making in the middle of a stable release cycle. But bugfixes can always break someone's code (I won't bother to link the obligatory xkcd on the topic). But it's just a bugfix, pure and simple, no excuses needed.

mbauman commented on HISTORY.md:
```diff
@@ -163,6 +163,10 @@ Standard library changes
   results table ([#38042]).
 * `@testset` now supports the option `verbose` to show the test result summary
   of the children even if they all pass ([#33755]).
+* In `LinearIndices(::Tuple)` and `CartesianIndices(::Tuple)`, integer elements (as opposed to ranges of integers) of the
+  argument tuple now always describe 1-based ranges, e.g, `CartesianIndices((3, 1:3))` is equivalent to
+  `CartesianIndices((1:3, 1:3))`. This is consistent with the (old and new) behavior for tuples containing only integers
+  and is considered a bugfix ([#37829], [#37928]).
```
How about this?  It's a little tricky to _both_ keep with the style of the NEWS (leading with the new behavior) and also lead with the bugfix (which — by definition — leads with the old behavior).  Here's a shot at walking down the middle:

```suggestion
* In `LinearIndices(::Tuple)` and `CartesianIndices(::Tuple)`, integers (as opposed to ranges of integers) in the
  argument tuple now consistently describe 1-based ranges, e.g, `CartesianIndices((3, 1:3))` is equivalent to
  `CartesianIndices((1:3, 1:3))`. This is how tuples of integers have always been documented to work, but a
  bug had caused erroneous behaviors with heterogeneous tuples containing both integers and ranges ([#37829], [#37928]).
```

sostock replied:
I like it.

sostock committed:
Rephrase

Following mbauman’s suggestion: https://github.com/JuliaLang/julia/pull/39784#discussion_r581033778

Co-authored-by: Matt Bauman <mbauman@gmail.com>

sostock commented:
This has been backported to 1.6-rc3, so this PR can be merged now.

mbauman commented:
572836ad82ae4b0ef792930f9772cc0dfddb8823
